python/ray/data/_internal/logical/optimizers.py

```diff
@@ -0,0 +1,42 @@
+from typing import List
+
+from ray.data._internal.execution.interfaces import PhysicalOperator
+from ray.data._internal.logical.interfaces import Rule, Optimizer, LogicalOperator
+from ray.data._internal.logical.planner import Planner
+
+
+class LogicalOptimizer(Optimizer):
+    """The optimizer for logical operators."""
+
+    @property
+    def rules(self) -> List[Rule]:
+        # TODO: Add logical optimizer rules.
+        return []
+
+
+class PhysicalOptimizer(Optimizer):
+    """The optimizer for physical operators."""
+
+    @property
+    def rules(self) -> List["Rule"]:
+        # TODO: Add physical optimizer rules.
+        return []
+
+
+class LogicalPlan:
+    """The plan with a DAG of logical operators."""
+
+    def __init__(self, dag: LogicalOperator):
+        self._dag = dag
```

Comment:
If operators are going to need to reference this `_dag` in order to to link it as an input operator, this should be exposed as part of `LogicalPlan`'s API.

```suggestion
        self._dag = dag
        
    @property
    def dag(self) -> LogicalOperator:
        return self._dag
```

Reply:
@clarkzinzow - added.